File: dashboard/frontend/src/types/person.ts
export interface Person {
  personId: number;
  fullName: string;
  nameVariations?: string[];
  primaryRole?: string;
  roles?: string[];
  emailAddresses?: string[];
  phoneNumbers?: string[];
  addresses?: string[];
  isRedacted: boolean;
  victimIdentifier?: string;
  dateOfBirth?: string;
  nationality?: string;
  occupation?: string;
  confidenceLevel: string;
  notes?: string;
  createdAt: string;
  updatedAt?: string;
  // Frequency counts
  documentCount?: number;
  eventCount?: number;
  relationshipCount?: number;
  financialCount?: number;
  totalMentions?: number;
  // Epstein relationship
  epsteinRelationship?: string;
}

export interface PersonDetail extends Person {
  relationshipCount?: number;
  eventCount?: number;
  documentCount?: number;
  financialTransactionCount?: number;
  mediaCount?: number;
}

export interface EntityFrequency {
  id: number;
  name: string;
  entityType: string;
  primaryRole?: string;
  documentCount: number;
  eventCount: number;
  relationshipCount: number;
  financialCount: number;
  financialTotal: number;
  mediaCount: number;
  totalMentions: number;
}

export interface DuplicateGroup {
  canonicalName: string;
  variants: Person[];
  totalDocuments?: number;
  totalEvents?: number;
  totalRelationships?: number;
}

File: dashboard/backend/src/EpsteinDashboard.Core/Entities/Person.cs
namespace EpsteinDashboard.Core.Entities;

public class Person
{
    public long PersonId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string? NameVariations { get; set; } // JSON
    public string? PrimaryRole { get; set; }
    public string? Roles { get; set; } // JSON
    public string? EmailAddresses { get; set; } // JSON
    public string? PhoneNumbers { get; set; } // JSON
    public string? Addresses { get; set; } // JSON
    public bool? IsRedacted { get; set; }
    public string? VictimIdentifier { get; set; }
    public string? DateOfBirth { get; set; }
    public string? Nationality { get; set; }
    public string? Occupation { get; set; }
    public long? FirstMentionedInDocId { get; set; }
    public string? ConfidenceLevel { get; set; }
    public string? Notes { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public Document? FirstMentionedInDocument { get; set; }
    public ICollection<Relationship> RelationshipsAsPerson1 { get; set; } = new List<Relationship>();
    public ICollection<Relationship> RelationshipsAsPerson2 { get; set; } = new List<Relationship>();
    public ICollection<EventParticipant> EventParticipations { get; set; } = new List<EventParticipant>();
    public ICollection<Communication> SentCommunications { get; set; } = new List<Communication>();
    public ICollection<CommunicationRecipient> ReceivedCommunications { get; set; } = new List<CommunicationRecipient>();
    public ICollection<FinancialTransaction> TransactionsAsFrom { get; set; } = new List<FinancialTransaction>();
    public ICollection<FinancialTransaction> TransactionsAsTo { get; set; } = new List<FinancialTransaction>();
    public ICollection<Location> OwnedLocations { get; set; } = new List<Location>();
    public ICollection<EvidenceItem> SeizedFromItems { get; set; } = new List<EvidenceItem>();
    public ICollection<MediaPerson> MediaAppearances { get; set; } = new List<MediaPerson>();
    public ICollection<VisualEntity> VisualIdentifications { get; set; } = new List<VisualEntity>();
    public ICollection<DocumentPerson> DocumentMentions { get; set; } = new List<DocumentPerson>();
}
